core/modules/replica/ChunkLocker.cc

```diff
@@ -0,0 +1,183 @@
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/ChunkLocker.h"
+
+// System headers
+#include <stdexcept>
+#include <tuple>        // std::tie
+
+// Qserv headers
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+///////////////////////////////////////
+//                Chunk              //
+///////////////////////////////////////
+
+bool Chunk::operator==(Chunk const& rhs) const {
+    return  std::tie(databaseFamily, number) ==
+            std::tie(rhs.databaseFamily, rhs.number);
+}
+
+bool Chunk::operator<(Chunk const& rhs) const {
+    return  std::tie(databaseFamily, number) <
+            std::tie(rhs.databaseFamily, rhs.number);
+}
+
+std::ostream& operator<<(std::ostream& os, Chunk const& chunk) {
+    os  << "Chunk (" << chunk.databaseFamily << ":" << chunk.number << ")";
+    return os;
+}
+
+/////////////////////////////////////////////
+//                ChunkLocker              //
+/////////////////////////////////////////////
+
+bool ChunkLocker::isLocked(Chunk const& chunk) const {
+    util::Lock mLock(_mtx, "ChunkLocker::isLocked(chunk)");
+    return _chunk2owner.count(chunk);
+}
+
+bool ChunkLocker::isLocked(Chunk const& chunk,
+                           std::string& owner) const {
+
+    util::Lock mLock(_mtx, "ChunkLocker::isLocked(chunk,owner)");
+
+    if (_chunk2owner.count(chunk)) {
+        owner = _chunk2owner.at(chunk);
+        return true;
+    }
+    return false;
+}
+
+ChunkLocker::ChunksByOwners ChunkLocker::locked(std::string const& owner) const {
+
+    util::Lock mLock(_mtx, "ChunkLocker::locked");
+
+    if (owner.empty()) return _owner2chunks;
+
+    ChunksByOwners owner2chunks;
+    if (_owner2chunks.count(owner)) {
```

Comment:
we dislike implicit conversion to bool

Reply:
There is a dedicated section in the LSST DM C+ Style Guide discussing the issue:
https://developer.lsst.io/cpp/style.html#implicit-test-for-0-should-not-be-used-other-than-for-boolean-variables

The last statement of that section says:  "...The latter [referring to implicit conversions to boolean[ is regarded as such a common practice in C/C++ however that it can be used...."

Frankly, I use this a lot (for counters and pointers) This makes code less noisy and more observable.